pkg/state/impl/inmem/collection.go

```diff
@@ -106,27 +107,29 @@ func (collection *ResourceCollection) inject(resource resource.Resource) {
 }
 
 // Create a resource.
-func (collection *ResourceCollection) Create(ctx context.Context, resource resource.Resource, owner string) error {
-	resource = resource.DeepCopy()
+func (collection *ResourceCollection) Create(ctx context.Context, res resource.Resource, owner string) error {
+	res = res.DeepCopy()
 
-	if err := resource.Metadata().SetOwner(owner); err != nil {
+	if err := res.Metadata().SetOwner(owner); err != nil {
 		return err
 	}
 
 	collection.mu.Lock()
 	defer collection.mu.Unlock()
 
-	if _, exists := collection.storage[resource.Metadata().ID()]; exists {
-		return ErrAlreadyExists(resource.Metadata())
+	if _, exists := collection.storage[res.Metadata().ID()]; exists {
+		return ErrAlreadyExists(res.Metadata())
 	}
 
+	res.Metadata().SetVersion(resource.NewVersion(1))
```

Comment:
we do `DeepCopy` as we store, but we should set the version before we do `DeepCopy`, or rather `DeepCopy` after we set the version, so that he `res` upon return from `Create` should contain correct `Metadata()`